File: Problems/FixTheCode/FTC_1_sorting-algorithms/python/source/bubbleSorter.py
from customSorter import CustomSorter
from arrays import swapTwoNumbers

class BubbleSorter(CustomSorter):

    def sort(self):
        print("Sorting array using Bubble Sort")

        moved = True
        while(moved):
            moved = False
            for i in range(len(self.array)):
                if (self.array[i] > self.array[i+1]):
                    swapTwoNumbers(self.array, i, i+1)
                    moved = True

File: Problems/FixTheCode/FTC_1_sorting-algorithms/java/source/BubbleSorter.java
public class BubbleSorter implements Sorter{

    public BubbleSorter(){}

    public int[] Sort(int[] array){
        System.out.println("Sorting array using Bubble Sort");
        int[] sortedArray = array.clone();
        int arrayLength = sortedArray.length;

        for (int i = 0; i < arrayLength; i++) {
            for (int j = 0; j < arrayLength - i; j++) {
                if (sortedArray[j] > sortedArray[j+1]){
                    Arrays.SwapTwoNumbers(sortedArray, j, j+1);
                }
            }
        }

       return sortedArray;
    }
}